server/upstream/download-pypi-versions

```diff
@@ -0,0 +1,259 @@
+                continue
+            pack_info = PackageInfo(*line.split(':', 3))
+            # skip python3-* lines
+            if pack_info.name.startswith('python3-'):
+                continue
+            # remove python- prefix
+            if pack_info.name.startswith('python-'):
+                pack_info.name = pack_info.name.replace('python-', '')
+            # now pack_info.name should have the original name from pypi
+            packages.append(pack_info)
+    return packages
+
+
+def __write_pypi_package_file(save_file, packages):
+    """write the pypi file. packages is a list of PackageInfo objs"""
+    with open(save_file, "w") as o:
+        for pi in sorted(packages):
+            if pi:
+                # FIXME(toabctl): check somehow if py2 and py3 versions are
+                # available currently just write python- and
+                # python3- names so both versions can be checked
+                o.write("%s:python-%s:%s:%s\n" %
+                        (pi.type, pi.name, pi.version, pi.url))
+                o.write("%s:python3-%s:%s:%s\n" %
+                        (pi.type, pi.name, pi.version, pi.url))
+
+
+def __create_pypi_package_file(save_file):
+    """create a new file with version information for pypi packages.
+    This step is expensive because it fetches the package list
+    from pypi (> 50000 packages) and then does a request for
+    every package to get the version information."""
+
+    # get current changelog serial and store in file before doing anything
+    # else so we can't lose some changes while we create the file
+    __write_cached_changelog_last_serial(save_file)
+
+    try:
+        pack_list = __get_package_list()
+        sys.stderr.write(
+            "Found %s packages. Getting packages details...\n" %
+            (len(pack_list)))
+        p = Pool(50)
+        packages = p.map(__get_package_info, pack_list)
+        __write_pypi_package_file(save_file, packages)
+    except Exception as e:
+        sys.stderr.write("Error while creating the initial pypi file: %s\n" % e)
+        # something wrong - delete the serial file so in the next run the
+        # the file will be recreated
+        serial_file = __get_save_file_serial_name(save_file)
+        os.remove(serial_file)
+    else:
+        sys.stderr.write("Initial pypi file '%s' created\n" % save_file)
+
+
+def __find_package_name_index(packages, name):
+    """find the given name in the packages list or None if not found"""
+    for i, pack in enumerate(packages):
+        if pack.name == name:
+            return i
+    return None
+
+
+def __update_pypi_package_file(save_file, current_changelog):
+    """update information of an exisiting file"""
+    # get current changelog serial and store in file before doing anything
+    # else so we can't lose some changes while we update the file
+    __write_cached_changelog_last_serial(save_file)
+
+    try:
+        packages = __read_pypi_package_file(save_file)
```

Comment:
> Maybe move this inside the if statement?

Done in https://github.com/toabctl/osc-plugin-collab/commit/2f370e3bf6fce74afc85f5de1f38382d091f9108